jr-annotation-support/src/main/java/com/fasterxml/jackson/jr/annotationsupport/AnnotationBasedValueRWModifier.java

```diff
@@ -32,4 +38,40 @@ public POJODefinition pojoDefinitionForSerialization(JSONWriter writeContext,
         return AnnotationBasedIntrospector.pojoDefinitionForSerialization(writeContext,
                 pojoType, _visibility);
     }
+
+    @Override
+    public ValueWriter overrideStandardValueWriter(JSONWriter writeContext, Class<?> type, int stdTypeId) {
+        if (stdTypeId == 23) {
+            return new EnumWriter(type);
+        }
+        return null;
+    }
+
+    private static class EnumWriter implements ValueWriter {
+        private final Class<?> _valueType;
+
+        public EnumWriter(Class<?> type) {
+            _valueType = type;
+        }
+
+        @Override
+        public void writeValue(JSONWriter context, JsonGenerator g, Object value) throws IOException {
+            // Try getting the value from the JsonProperty annotation, if it fails just use the enum name.
```

Comment:
This introspection should be done once when constructing writer, and not each every time value is written -- introspection access is pretty slow.